nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1662,6 +1663,12 @@ def setup(self, stage=None):
             # Override limit_val_batches to be a multiple of num microbatches to prevent val_step from exiting in between a step
             self._reconfigure_limit_batches(self.trainer.limit_val_batches, self._validation_dl, 'val')
 
+        # Data cache generation only
+        # Stops training after creating a data cache
+        if self.cfg.data.get('data_cache_generation_only', False):
+            self.trainer.num_sanity_val_steps = 0
+            self.trainer.should_stop = True
+
```

Comment:
@dimapihtar training wouldn't have started at all right to stop it here ? Since we are checking in the setup() method which is run before training begins.

Reply:
> @dimapihtar training wouldn't have started at all right to stop it here ? Since we are checking in the setup() method which is run before training begins.

yes, but trainer is already initialized at that moment so it quits the scripts.

Reply:
@athitten it quits the script before the first training step.

Reply:
Thanks @dimapihtar, the comment `# Stops training after creating a data cache` made me think training would have started. So with `data_cache_generation_only` arg we just cache data and don't run or start any training at all ?

Reply:
@athitten yes, exactly. Probably not the best explanation from my side.